File: 6472.SumOfMatrixAfterQueries/passedSolution.java
public class passedSolution {
    public long matrixSumQueries(int n, int[][] q) {
        boolean[] rows = new boolean[n], cols = new boolean[n];
        long ans = 0;
        int remainRows = n, remainCols = n;
        for (int i = q.length; i >= 0; i--) {
            int type = q[i][0], index = q[i][1], v = q[i][2];
            if (type == 0 && !rows[index]) {
                rows[index] = true;
                ans += v * remainCols;
                remainRows--;
            } else if (type == 1 && !cols[index]) {
                cols[index] = true;
                ans += v * remainRows;
                remainCols--;
            }
        }
        return ans;
    }
}

File: 6472.SumOfMatrixAfterQueries/passedSolution.py
from typing import List


class Solution:
    def matrixSumQueries(self,n:int,q:List[List])->int:
        remainCols,remainRows,ans = n ,n,0
        rows,cols = [False] * n, [False] * n
        for i in range(len(q) - 1, -1 , -1):
            t,index, v = q[i][0], q[i][1], q[i][2] 
            if t == 0 and not rows[index]:
                rows[index] = True
                ans += remainCols * v
                remainRows -=1 
            elif t == 1 and not cols[index]:
                cols[index] = True
                ans+= remainRows * v
                remainCols -= 1

        return ans;
